addons/account_edi/models/account_edi_document.py

```diff
@@ -222,3 +216,31 @@ def _process_documents_web_services(self, job_count=None, with_commit=True):
             else:
                 if with_commit and len(jobs) > 1:
                     self.env.cr.commit()
+
+    def _process_documents_no_web_services(self):
+        """ Post and cancel all the documents that don't need a web service.
+        """
+        jobs = self.filtered(lambda d: not d.edi_format_id._needs_web_services())._prepare_jobs()
+        for documents, doc_type in jobs:
+            self._process_job(documents, doc_type)
+
+    def _process_documents_web_services(self):
+        ''' Post and cancel all the documents that need a web service. '''
+        jobs = self.filtered(lambda d: d.edi_format_id._needs_web_services())._prepare_jobs()
+        self._process_jobs(jobs, with_commit=False)
+
+    @api.model
+    def _cron_process_documents_web_services(self, job_count=None):
+        ''' Method called by the EDI cron processing all web-services.
+
+        :param job_count: Limit explicitely the number of web service calls. If not provided, process all.
+        '''
+        edi_documents = self.search([('state', 'in', ('to_send', 'to_cancel'))])
+        all_jobs = edi_documents.filtered(lambda d: d.edi_format_id._needs_web_services())._prepare_jobs()
+        jobs_to_process = all_jobs[0:job_count] if job_count else all_jobs
+
+        self._process_jobs(jobs_to_process)
+
+        # Mark the CRON to be triggered again asap since there is some remaining jobs to process.
+        if len(all_jobs) - len(jobs_to_process):
+            self.env.ref('account_edi.ir_cron_edi_network')._trigger()
```

Comment:
This is new, we re-trigger the cron inside the cron. The purpose is to be nice with others by processing some records, give back the worker so it can process other jobs then come back to process some more. For those familiar with async programming in python, it is equivalent to the bellow async/await code:

```py
async def cpu_intensive_task():
    for i in range(1000)
        if i % 100 == 0:
            await asyncio.sleep(0)  # be nice to others

        time.sleep(1)  # simulate a cpu intensive task
```